dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNPhaseTapChangerInterfaceIIDM.cpp

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNPhaseTapChangerInterfaceIIDM.cpp
+ *
+ * @brief Phase tap changer data interface : implementation file for IIDM implementation
+ *
+ */
+
+#include "DYNPhaseTapChangerInterfaceIIDM.h"
+
+#include "DYNStepInterfaceIIDM.h"
+
+#include <powsybl/iidm/TapChanger.hpp>
+
+using boost::shared_ptr;
+
+namespace DYN {
+
+PhaseTapChangerInterfaceIIDM::PhaseTapChangerInterfaceIIDM(powsybl::iidm::PhaseTapChanger& tapChanger) : tapChangerIIDM_(tapChanger) {
+  auto oldTapPosition = tapChanger.getTapPosition();
+  for (long i = tapChanger.getLowTapPosition(); i <= tapChanger.getHighTapPosition(); i++) {
+    tapChanger.setTapPosition(i);
```

Comment:
This code is not in the original class

Reply:
Yes, original lacks initializing the steps_ field. I propose a corrected code. A better code will be to make step_ pointing to its equivalent vector ot the powsybl object. I did not find it (for now).

Reply:
This is because it is done somewhere else, please remove this

Reply:
OK. I have to put it in the test program; it's not where it belongs. I will push within the day for that change.